client/layout/transient-notices/snackbar/index.js

```diff
@@ -40,19 +40,42 @@ function Snackbar(
 		actions = [],
 		onRemove = noop,
 		icon = null,
+		explicitDismiss = false,
+		// onDismiss is a callback executed when the snackbar is dismissed.
+		// It is distinct from onRemove, which _looks_ like a callback but is
+		// actually the function to call to remove the snackbar from the UI.
+		onDismiss = null,
```

Comment:
This is confusing... how about we rename `onRemove` to something like `removeSnackbar` or `removeFunction`?

Reply:
I agree this is confusing, I wanted to rename `onRemove` but I'm concerned about backwards compatibility - this is to be merged upstream to Gutenberg

Reply:
Ah, right. That makes sense.